connectors/runner.py

```diff
@@ -180,29 +180,20 @@ async def poll(self, one_sync=False, sync_now=True):
             while self.running:
                 try:
                     logger.debug(f"Polling every {self.idling} seconds")
-                    async for connector in self.connectors.get_list():
-                        if (
-                            connector.service_type not in native_service_types
-                            and connector.id not in connectors_ids
-                        ):
-                            logger.debug(
-                                f"Connector {connector.id} of type {connector.service_type} not supported, ignoring"
-                            )
-                            continue
-
+                    async for connector in self.connectors.get_connectors(
+                        native_service_types, connectors_ids
+                    ):
                         await self._one_sync(connector, es, sync_now)
                         if one_sync:
-                            self.stop()
                             break
                 except Exception as e:
                     logger.critical(e, exc_info=True)
                     self.raise_if_spurious(e)
 
                 if not one_sync:
                     await self._sleeps.sleep(self.idling)
-                else:
-                    self.stop()
         finally:
+            self.stop()
```

Comment:
this breaks `--one-sync`

Reply:
In line 188, it breaks out of the loop if it's `one_sync`, I just moved `self.stop()` to `finally`, as I think this should be run as a cleanup for all the scenarios.

Reply:
Thanks, So, the `break` here does not change the `running` flag. We are going to stay in `while self.running` -- I assume this will be fixed in the other patch ?

Reply:
Updated in both https://github.com/elastic/connectors-python/pull/219/commits/15e8bd33c6f4223a5a90f81ed9d8c141db0d0d6a and https://github.com/elastic/connectors-python/pull/218/commits/9733f1754fbae723b9ce95833254379ce9f3ac0f